lib/cretonne/src/ir/types.rs

```diff
@@ -35,6 +35,22 @@ pub const VOID: Type = Type(0);
 // 512-bit SIMD vectors.
 include!(concat!(env!("OUT_DIR"), "/types.rs"));
 
+/// Possible derivation functions for types. Used in runtime typechecking
+pub enum DerivedFunctions {
```

Comment:
It isn't neccessary. I added it just to simplify the work of gen_legalizer.py in the face of possible over/underflow in the "derived" types we are comparing. I removed it in 
29b987ac8713a8a8764bf2f26a8fac8a62b6f4dc